### PR TITLE
[Support] Code cleanup and refactor how b-panel animation occurs [#157398629]

### PR DESCRIPTION
**Pull request by jonpitch**
## Overview
This is just some cleanup to facilitate [this](https://www.pivotaltracker.com/story/show/157398629) story

## How to test
See impact app PR

### CODE REVIEW

File: addon/components/b-panel-content.js [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
 export default Component.extend({
   layout,
   tagName: 'div',
   classNames: ['b-panel-content'],
   open: false,
-  closed: computed('open', function(){
-    return !get(this, 'open');
-  }),
-  classNameBindings: ['open', 'closed'],
+  classNameBindings: ['open:open:closed'],
```
Comment (kikemoto) [inline addon/components/b-panel-content.js]: @jonpitch Why was this changed from `classNameBindings: ['open', 'closed']`? I'm familiar with `[property:classname]` to map a property to a customized classname, but not familiar with the outcome of `['open:open:closed']`.

Comment (jonpitch) [reply addon/components/b-panel-content.js]: @kikemoto 

with this:
```javascript
classNameBindings: ['open', 'closed']
```

when the value of the attribute `open = true`, the class of `'open'` is applied to the element. When `closed = true`, the class of `'closed'` is applied to the element.

with:
```javascript
classNameBindings: ['open:open:closed']
```

If `open = true`, the the class `'open'` is applied, else `'closed'` is applied. 

They are the same thing, just expressed differently. I have no preference which is used, so happy to revert if the original is more clear. There's a better overview [here](https://guides.emberjs.com/v3.1.0/components/customizing-a-components-element/#toc_customizing-the-element-s-class) which has some other ways you can write the same thing.

**Comment by thematthopkins** [comment]
@jonpitch I'm a little curious about the `observer` -> `computedProperty` change.  `observer` seems to make sense, because it's there to perform side-effects on the DOM.  A `computedProperty` that is impure + performs side-effects seems counter intuitive, and I'd be surprised if there are any guarantees around the min/max number of times `computedProperty` gets evaluated under any condition.  Is a `computedProperty` better here for some reason?

**Comment by jonpitch** [comment]
@thematthopkins in my opinion, the majority of the time `observer` is used in our Ember apps, it has unintended side effects and the same functionality can be achieved simpler. In this particular case, `observer` probably makes more sense, but I'd rather have the rule to not use them globally thinking it'll have more benefits even if it makes this one instance a little more difficult.
